hbase-server/src/main/java/org/apache/hadoop/hbase/wal/WALEdit.java

```diff
@@ -257,30 +317,55 @@ public static WALEdit createFlushWALEdit(RegionInfo hri, FlushDescriptor f) {
   }
 
   public static FlushDescriptor getFlushDescriptor(Cell cell) throws IOException {
-    if (CellUtil.matchingColumn(cell, METAFAMILY, FLUSH)) {
-      return FlushDescriptor.parseFrom(CellUtil.cloneValue(cell));
-    }
-    return null;
+    return CellUtil.matchingColumn(cell, METAFAMILY, FLUSH)?
+        FlushDescriptor.parseFrom(CellUtil.cloneValue(cell)): null;
   }
 
+  /**
+   * @return A meta Marker WALEdit that has a single Cell whose value is the passed in
+   * <code>regionEventDesc</code> serialized and whose row is this region,
+   * columnfamily is {@link #METAFAMILY} and qualifier is
+   * {@link #REGION_EVENT_PREFIX} + {@link RegionEventDescriptor#getEventType()};
+   * for example HBASE::REGION_EVENT::REGION_CLOSE.
+   */
   public static WALEdit createRegionEventWALEdit(RegionInfo hri,
       RegionEventDescriptor regionEventDesc) {
-    KeyValue kv = new KeyValue(getRowForRegion(hri), METAFAMILY, REGION_EVENT,
+    KeyValue kv = new KeyValue(getRowForRegion(hri), METAFAMILY,
+        createRegionEventDescriptorQualifier(regionEventDesc.getEventType()),
```

Comment:
Maybe we could pre-create all the region events?

Reply:
Could. My thinking is these event types are super rare in the scheme of things so shouldn't be a concern?